### ISSUE TITLE
bulk: increased split-after size can result in replica imbalance

### ISSUE THREAD

**Issue by nicktrav** [cockroachdb/cockroach#75664]
**Describe the problem**

Spin-off from #68303 - provided a distilled version. Context on original, [here](https://github.com/cockroachdb/cockroach/issues/68303#issuecomment-1024526795).

It appears as though e12c9e65c457e035562e71621d164f04abb33728 causes replica imbalance which doesn't play nicely with large imports.

With that commit, on `clearrange/checks=true`, the import fails as a number of nodes run out of disk due to the imbalance:

<img width="1005" alt="Screen Shot 2022-01-28 at 9 37 24 AM" src="https://user-images.githubusercontent.com/804457/151609265-62217a09-5873-44d0-b689-7b6447cc8407.png">

Without that commit the import succeeds:

<img width="1005" alt="Screen Shot 2022-01-28 at 10 47 39 AM" src="https://user-images.githubusercontent.com/804457/151609284-2a7eb9c4-39e2-4d97-989b-9cfec52ddbcc.png">

Jira issue: CRDB-12773

**Comment by nicktrav** [comment]
cc: @dt

**Comment by dt** [comment]
Reverting in https://github.com/cockroachdb/cockroach/pull/75882 for now, will try to find some time next week to poke at this, and see why the closer-to-default split size does so badly / what the extremely low split size was better for the allocator. Maybe the bigger range counts were covering for scatter's known padding issues? Might need @aayushshah15 to help me poke at this when I get to it.

**Comment by aayushshah15** [comment]
>Maybe the bigger range counts were covering for scatter's known padding issues? Might need @aayushshah15 to help me poke at this when I get to it.

I'd still expect the allocator to fix straight-up replica count divergence though. Very curious why this was happening. Feel free to put something on my calendar if you want to look together, but I can also just try repro-ing on my own with the split size bumped.

**Comment by aayushshah15** [comment]
I am playing with this a little bit (i.e. running with `384mb` with [our patch for `AdminScatter`](https://github.com/cockroachdb/cockroach/pull/75894)) on a tpcc import. Repro steps are identical to what I've described in that PR, except for the increased split-after size.

The main things that I've noticed so far are that, with `384mb`, we essentially stop scattering after the very beginning of the import. My sense is that this runs contrary to our understanding (also something that wasn't happening with `48mb`)? Yellow is rebalances.
![image](https://user-images.githubusercontent.com/10788754/152370256-e579a8f8-cb82-4092-927c-23968fc2b142.png)

Additionally, as far as I can tell, the allocator is continuously rebalancing. The rate at which we're converging is just slower than the rate at which newer splits are being created. After the import is done, I'd expect this to fully converge.

```
        {
          "time": "2022-02-03T15:12:34.421667145Z",
          "message": "kv/kvserver/allocator_scorer.go:186 [n7,status] s7: should-rebalance(ranges-overfull): rangeCount=1909, mean=1291.00, overfull-threshold=1356"
        },
```
![image](https://user-images.githubusercontent.com/10788754/152371957-27e13633-7a71-4f4f-a1fc-d871ff28054a.png)
